File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        
        for (int i = 0; i < photo.length; i++) {
            int number = 0;
            for (String item : photo[i]) {
                for (int j = 0; j < name.length; j++) {
                    if (name[j].equals(item)) {
                        number += yearning[j];
                    }
                }
            }
            answer[i] = number;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    for(let i = 0; i < photo.length; i++){
        var number = 0;
        photo[i].filter((item)=>{
            name.map((list,index)=>{
                if(list === item){
                    number += yearning[index]
                }
            })
        })
        answer.push(number)
    }
    
    return answer;
}